src/main/java/pt/tqs/g205/services/PratoService.java

```diff
@@ -3,27 +3,71 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import pt.tqs.g205.domain.IngredientesPorPrato;
 import pt.tqs.g205.domain.Prato;
+import pt.tqs.g205.repositories.IngredientesPorPratoRepository;
 import pt.tqs.g205.repositories.PratoRepository;
 
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
 
 /**
- * Servicos para a classe Prato.
+ * Logica de servicos para os pratos.
  */
 @Service
 public class PratoService {
 
   @Autowired
   private PratoRepository pratoRepo;
+
+  @Autowired
+  private IngredientesPorPratoRepository ingredientesPorPratoRepo;
   
   /**
    * Obter todos os pratos.
-   * @return lista de Pratos.
+   * @return lista de todos os pratos.
    */
   public List<Prato> getAll() {
-    return pratoRepo.findAll();
+    List<Prato> pratos = pratoRepo.findAll();
+
+    pratos.forEach(e -> {
+      e.setIngredientes(null);
+    });
+
+    return pratos;
+  }
+  
+  /**
+   * Obter um prato por id.
+   * @param id id do prato.
+   * @return prato filtrado.
+   */
+  public Prato getPratoById(Integer id) {
+    Prato prato = pratoRepo.getOne(id);
+
+    if (prato == null) {
+      throw new NoSuchElementException();
+    }
+    
+    List<IngredientesPorPrato> ingredientesPorPrato = ingredientesPorPratoRepo.findByPratoId(id);
+
+    prato.setIngredientes(ingredientesPorPrato);
+
+    return prato;
+  }
+  
+  /**
+   * Obter pratos pelo numero de calorias.
+   * @param value valor maximo de calorias.
+   * @return lista de pratos filtrados.
+   */
+  public List<Prato> getByCalorias(Double value) {
+    List<Prato> pratos = pratoRepo.findAll();
 
+    List<Prato> res =
+        pratos.stream().filter(e -> e.getCalorias() <= value).collect(Collectors.toList());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Immediately return this expression instead of assigning it to the temporary variable "res". ([squid:S1488](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS1488))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=6&issues=AWNEkG2rXOBV_Yrh_0kh&open=AWNEkG2rXOBV_Yrh_0kh)

[ISSUE_KEY]: AWNEkG2rXOBV_Yrh_0kh